plugins/InspectionGadgets/src/com/siyeh/ig/redundancy/RedundantStringOperationInspection.java

```diff
@@ -131,6 +134,55 @@ public void visitMethodCallExpression(@NotNull PsiMethodCallExpression call) {
       myProcessors.mapAll(call).forEach(myHolder::registerProblem);
     }
 
+    @Override
+    public void visitBinaryExpression(@NotNull PsiBinaryExpression expression) {
+      super.visitBinaryExpression(expression);
+      if (!ComparisonUtils.isComparison(expression)) {
+        return;
+      }
+      final OrderedBinaryExpression<PsiMethodCallExpression, PsiExpression>
+        orderedBinaryExpression = OrderedBinaryExpression.from(expression, PsiMethodCallExpression.class);
+      if (orderedBinaryExpression == null) {
+        return;
+      }
+      final PsiMethodCallExpression lengthCallExpression = orderedBinaryExpression.getFirstOperand();
+      if (!STRING_LENGTH.test(lengthCallExpression)) {
+        return;
+      }
+      final PsiReferenceExpression methodExpression = lengthCallExpression.getMethodExpression();
+      final PsiExpression qualifierExpression = methodExpression.getQualifierExpression();
+      if (!(qualifierExpression instanceof PsiMethodCallExpression stripCallExpression)) {
+        return;
+      }
+      if (!STRIP.test(stripCallExpression)) {
+        return;
+      }
+      String message = InspectionGadgetsBundle.message("expression.can.be.replaced.problem.descriptor", "isBlank()");
+      final Object object = ExpressionUtils.computeConstantExpression(orderedBinaryExpression.getSecondOperand());
+      if (!(object instanceof Integer) && !(object instanceof Long) && !(object instanceof Short)) {
```

Comment:
I believe, it cannot be Short. And I'm not sure whether it's good to issue a warning when there's a non-trivial constant computation. Let's stay with literals like 0 and 1 only.

Reply:
We also have com.siyeh.ig.psiutils.JavaPsiMathUtil#getRangeFromComparison which can help here. It will return [1..MAX_VALUE] for `>= 1` and `>0`. You can intersect the result with non-negative numbers (JvmPsiRangeSetUtil#indexRange), then check if the result is exactly 0, or it's 1..MAX_VALUE.

Reply:
1. Done
2. I think it's not usable here. I think it works only with a previous `if` statement, not in a boolean expression. I think it will return `[MIN_VALUE..MAX_VALUE]`.